src/print/node-helpers.ts

```diff
@@ -10,17 +10,53 @@ import {
     SlotNode,
     TitleNode,
     WindowNode,
+    IfBlockNode,
+    AwaitBlockNode,
+    CatchBlockNode,
+    EachBlockNode,
+    ElseBlockNode,
+    KeyBlockNode,
+    PendingBlockNode,
+    ThenBlockNode,
 } from './nodes';
 import { inlineElements, TagName } from '../lib/elements';
 import { FastPath } from 'prettier';
-import { isASTNode } from './helpers';
+import { findLastIndex, isASTNode, isPreTagContent } from './helpers';
 
 const unsupportedLanguages = ['coffee', 'coffeescript', 'pug', 'styl', 'stylus', 'sass'];
 
 export function isInlineElement(node: Node) {
     return node.type === 'Element' && inlineElements.includes(node.name as TagName);
 }
 
+export function isBlockElement(path: FastPath, node: Node): node is ElementNode {
+    // TODO switch to a list of tags instead
```

Comment:
The idea is to switch to a defined list of block elements instead of a defined set of inline elements.